### DISCUSSION TITLE
Issue - Unable to start command server. - ulimit issue

### DISCUSSION THREAD

**Discussion by pavandonthireddy** [gnzsnz/ib-gateway-docker#87] [Q&A] [answered]
Hi - I'm trying to run the latest image with default settings in a linux VM. However I get the below message saying that unable to start Command server. Any idea what is happening here ?

ib-gateway-1  | 2024-04-01 21:07:04:629 IBC: using default settings provider: ini file is /home/ibgateway/ibc/config.ini

ib-gateway-1  | 2024-04-01 21:07:04:629 IBC: using default login manager: will get username and password from settings; FIX username and password (if required) from settings

ib-gateway-1  | 2024-04-01 21:07:04:629 IBC: using default main window manager

ib-gateway-1  | 2024-04-01 21:07:04:630 IBC: using default trading mode manager: constructor parameter args but trading mode not present - will be taken from settings

ib-gateway-1  | 2024-04-01 21:07:04:632 IBC: using default config dialog manager

ib-gateway-1  | 2024-04-01 21:07:04:644 IBC: CommandServer is not started because the port is not configured

ib-gateway-1  | library initialization failed - unable to allocate file descriptor table - out of memory/home/ibgateway/ibc/scripts/ibcstart.sh: line 504:    33 Aborted                 (core dumped) "$java_path/java" -cp "$ibc_classpath" $java_vm_options$autorestart_option $entry_point "$ibc_ini" ${mode}

ib-gateway-1  | IBC returned exit status 134

ib-gateway-1  | autorestart file not found

ib-gateway-1  | 

ib-gateway-1  | Gateway finished

ib-gateway-1  | 

ib-gateway-1 exited with code 0

**Comment by gnzsnz** [comment]
the message on the log stating "IBC: CommandServer" is not an error. it's just stating that commandserver will not be started.

the line bellow stating that you don't have enough memory. that looks like an error.

check that you VM has enough memory.

**Reply by eyalco** [reply]
I get the same error of not enough memory. I used to run ib-gateways on this vps which had enough memory before. I get the exact error trace as the question asked.

**Reply by gnzsnz** [reply]
hard to say without any details, you need at least ~768Mb+stack memory + what OS needs, I would say 2Gb to be on the safe side, TWS recommends 4Gbs. I can't find a recommendation for ibgateway.

A full log might be helpfull

**Reply by eyalco** [reply, marked as answer]
Well I searched around and found a solution. The problem is not the total memory of the docker but the file descriptor allocation which can be set by adding the following paramters in the docker-compose service:
   ulimits:
      nofile:
        soft: 65536
        hard: 65536
for example. this solved the issue for me

**Reply by eyalco** [reply]
the indentation was not kept in my comment use proper indentation for the docker compose syntax

**Reply by gnzsnz** [reply]
this seems to explain the issue you are facing

https://stackoverflow.com/questions/68776387/docker-library-initialization-failed-unable-to-allocate-file-descriptor-tabl

**Reply by GonzRon** [reply]
thanks, I just hit this with Fedora 41 - apparently the default systemd service file does configure the dockerd daemon to start with infinite fd limits

**Comment by gnzsnz** [comment]
related to #327 

setting ulimits in docker-compose solves the issue, https://docs.docker.com/reference/compose-file/services/#ulimits

```yaml
ulimits:
  nofile:
    soft: 65536 # increase to a suitable value
    hard: 65536 # increase to a suitable value
```

For reference my OS default of 1024 for "nofile" is sufficient for TWS and ib-gateway
